src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -528,26 +528,31 @@ std::unique_ptr<Uptane::Target> SotaUptaneClient::findTargetInDelegationTree(con
   return findTargetHelper(*toplevel_targets, target, 0, false);
 }
 
-result::Download SotaUptaneClient::downloadImages(const std::vector<Uptane::Target> &targets,
+result::Download SotaUptaneClient::downloadImages(std::vector<Uptane::Target> targets,
                                                   const api::FlowControlToken *token) {
   // Uptane step 4 - download all the images and verify them against the metadata (for OSTree - pull without
   // deploying)
   std::lock_guard<std::mutex> guard(download_mutex);
   result::Download result;
   std::vector<Uptane::Target> downloaded_targets;
-  for (auto it = targets.cbegin(); it != targets.cend(); ++it) {
-    auto images_target = findTargetInDelegationTree(*it);
+  for (auto &it : targets) {
```

Comment:
When using this style of `for`, it makes it clearer to rename the variable to something other than `it`, as it's not an iterator anymore.

Reply:
Good call. Done.